joomla/joomla-cms
# Support Codemirror's included key mappings

### Summary of Changes
First of all, this includes CodeMirror 5.35.0 so it's the same as #19809
Additionally, it removes the old option for "Vim Mode" and replaces it with additional choices for key mappings. Vim is still one of them (and will be used if you had previously selected Vim Mode) but now Emacs and Sublime Text are also provided. So, if you're familiar with any of those editors, you can make CodeMirror use similar keys and behave in mostly the same way. 

Since it doesn't make much sense to load all key mappings at the same time (you use either zero or one of them), key mapping info will not be loaded unless needed. As a result, the addons file (a bundle of several codemirror plugins) is much smaller now.

### Testing Instructions
Use CodeMirror. If you are familiar with Vim, Emacs, or Sublime Text, use one of those key mappings. Notice the change in CodeMirror's behavior. 


### Expected result
CodeMirror should work as usual. Additional Key Mapping modes should work somewhat like other editors.


### Actual result
As expected.


### Documentation Changes Required
I guess not.
